@zhutik 

`
boolean.go:12:28: exported func NewStringBool returns unexported type *adyen.stringBool, which can be annoying to use
Found 2 lint suggestions; failing.
`

--- a/boolean.go
+++ b/boolean.go
@@ -9,11 +9,13 @@ import (
 // which appear as strings instead of bools.
 type StringBool bool
 
+// NewStringBool returns an instance of StringBool representing a given bool
 func NewStringBool(b bool) *StringBool {
 	sb := StringBool(b)
 	return &sb
 }
 
+// UnmarshalJSON unmarshalls to a StringBool from a slice of bytes
 func (b *StringBool) UnmarshalJSON(data []byte) (err error) {
 	str := strings.TrimFunc(strings.ToLower(string(data)), func(c rune) bool {
 		return c == ' ' || c == '"'
@@ -28,6 +30,7 @@ func (b *StringBool) UnmarshalJSON(data []byte) (err error) {
 	return
 }
 
+// MarshalJSON marshalls a StringBool to a slice of bytes
 func (b StringBool) MarshalJSON() ([]byte, error) {
 	boolResult := bool(b)
 	var boolString string

--- a/boolean.go
+++ b/boolean.go
@@ -5,16 +5,16 @@ import (
 	"strings"
 )
 
-// stringBool allows us to unmarhsal Adyen Boolean values
+// StringBool allows us to unmarhsal Adyen Boolean values
 // which appear as strings instead of bools.
-type stringBool bool
+type StringBool bool
 
-func NewStringBool(b bool) *stringBool {
-	sb := stringBool(b)
+func NewStringBool(b bool) *StringBool {
+	sb := StringBool(b)
 	return &sb
 }
 
-func (b *stringBool) UnmarshalJSON(data []byte) (err error) {
+func (b *StringBool) UnmarshalJSON(data []byte) (err error) {
 	str := strings.TrimFunc(strings.ToLower(string(data)), func(c rune) bool {
 		return c == ' ' || c == '"'
 	})
@@ -24,11 +24,11 @@ func (b *stringBool) UnmarshalJSON(data []byte) (err error) {
 		return
 	}
 
-	*b = stringBool(parsed)
+	*b = StringBool(parsed)
 	return
 }
 
-func (b stringBool) MarshalJSON() ([]byte, error) {
+func (b StringBool) MarshalJSON() ([]byte, error) {
 	boolResult := bool(b)
 	var boolString string
 

--- a/boolean_test.go
+++ b/boolean_test.go
@@ -6,11 +6,11 @@ import (
 )
 
 type thing struct {
-	Bool stringBool `json:"value"`
+	Bool StringBool `json:"value"`
 }
 
 type thingWithEmpty struct {
-	Bool *stringBool `json:"value,omitempty"`
+	Bool *StringBool `json:"value,omitempty"`
 }
 
 func TestStringBool_Unmarshal(t *testing.T) {
@@ -84,7 +84,7 @@ func TestStringBool_Unmarshal(t *testing.T) {
 				t.Fatalf("expected error but didn't get one")
 			}
 
-			if stringBool(c.exp) != th.Bool {
+			if StringBool(c.exp) != th.Bool {
 				t.Fatalf("my exp: %v but got %v", c.exp, th.Bool)
 			}
 		})
@@ -99,12 +99,12 @@ func TestStringBool_Marshal(t *testing.T) {
 	}{
 		{
 			name:     "true",
-			object:   thing{Bool: stringBool(true)},
+			object:   thing{Bool: StringBool(true)},
 			expected: `{"value":"true"}`,
 		},
 		{
 			name:     "false",
-			object:   thing{Bool: stringBool(false)},
+			object:   thing{Bool: StringBool(false)},
 			expected: `{"value":"false"}`,
 		},
 	}
@@ -122,8 +122,8 @@ func TestStringBool_Marshal(t *testing.T) {
 }
 
 func TestStringBool_MarshalWithOmitempty(t *testing.T) {
-	valueTrue := stringBool(true)
-	valueFalse := stringBool(false)
+	valueTrue := StringBool(true)
+	valueFalse := StringBool(false)
 
 	cases := []struct {
 		name     string

--- a/notification.go
+++ b/notification.go
@@ -6,7 +6,7 @@ import "time"
 //
 // Link - https://docs.adyen.com/developers/api-reference/notifications-api#notificationrequest
 type NotificationRequest struct {
-	Live              stringBool                `json:"live"`
+	Live              StringBool                `json:"live"`
 	NotificationItems []NotificationRequestItem `json:"notificationItems"`
 }
 
@@ -40,5 +40,5 @@ type NotificationRequestItemData struct {
 	OriginalReference   string     `json:"originalReference,omitempty"`
 	PaymentMethod       string     `json:"paymentMethod"`
 	Reason              string     `json:"reason,omitempty"`
-	Success             stringBool `json:"success"`
+	Success             StringBool `json:"success"`
 }

--- a/payment.go
+++ b/payment.go
@@ -77,7 +77,7 @@ type AdditionalData struct {
 	Content       string      `json:"card.encrypted.json,omitempty"`
 	AliasType     string      `json:"aliasType,omitempty"`
 	Alias         string      `json:"alias,omitempty"`
-	ExecuteThreeD *stringBool `json:"executeThreeD,omitempty"`
+	ExecuteThreeD *StringBool `json:"executeThreeD,omitempty"`
 }
 
 // BrowserInfo hold information on the user browser

